sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -67,24 +67,38 @@ def csr_mean_variance_axis0(X):
         Feature-wise variances
 
     """
-    cdef unsigned int n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
-
-    cdef np.ndarray[DOUBLE, ndim=1, mode="c"] X_data
-    X_data = np.asarray(X.data, dtype=np.float64)     # might copy!
-    cdef np.ndarray[int, ndim=1] X_indices = X.indices
+    if X.dtype == np.int32:
```

Comment:
And is there a reason we don't support [other integer sparse matrix types](https://github.com/scipy/scipy/blob/master/scipy/sparse/_csparsetools.pyx.in#L16) (including `bool`, although mean and variance are silly functions to run in that case)? If the code supported those dtypes before (even if untested), I think we'd better maintain support.